python/pyspark/sql/datasource.py

```diff
@@ -183,11 +186,40 @@ def streamWriter(self, schema: StructType, overwrite: bool) -> "DataSourceStream
             message_parameters={"feature": "streamWriter"},
         )
 
+    def _streamReader(self, schema: StructType) -> "DataSourceStreamReader":
```

Comment:
Why do we need this `_streamReader` in datasource API?

Reply:
This is a private method to fall back to simple reader when streaming reader is not implemented. I would be ok to change this if there is an alternative.